gpt_index/indices/vector_store/vector_indices.py

```diff
@@ -367,3 +371,53 @@ def __init__(
             vector_store=vector_store,
             **kwargs,
         )
+
+
+class GPTOpensearchIndex(GPTVectorStoreIndex):
+    """GPT Opensearch Index
+
+    The GPTOpensearchIndex is a data structure where nodes are keyed by
+    embeddings, and those embeddings are stored in a document that is indexed with its embedding as well as
+    its textual data (text field is defined in the OpensearchVectorClient).
+    During index construction, the document texts are chunked up,
+    converted to nodes with text; each node's embedding is computed, and then
+    the node's text, along with the embedding, is converted into JSON document that
+    is indexed in Opensearch. The embedding data is put into a field with type "knn_vector" and the text
+    is put into a standard Opensearch text field.
+
+    During query time, the index performs approximate KNN search using the "knn_vector" field
+    that the embeddings were mapped to.
+
+    Args:
+        text_qa_template (Optional[QuestionAnswerPrompt]): A Question-Answer Prompt
+            (see :ref:`Prompt-Templates`).
+        endpoint (str): Endpoint of the elasticsearch cluster (required).
+        index_name (str): Index in the elasticsearch cluster (required).
+        embedding_field (str): Field (within the JSON document) that the embeddings are stored in.
+        embed_model (Optional[BaseEmbedding]): Embedding model to use for
```

Comment:
let's add OpensearchVectorClient to docstring as well

Reply:
Oops yeah this docstring is totally out of date, forgot to update this.